pkg/config/tlscfg/cert_watcher_test.go

```diff
@@ -82,32 +82,36 @@ func TestReload(t *testing.T) {
 	require.NoError(t, err)
 	assert.Equal(t, &cert, watcher.certificate())
 
-	// update the content with client certs
+	// Write the client's public key.
 	certData, err = ioutil.ReadFile(clientCert)
 	require.NoError(t, err)
 	err = syncWrite(certFile.Name(), certData, 0644)
 	require.NoError(t, err)
+
+	waitUntil(func() bool {
+		// Logged when the cert is reloaded with mismatching client public key and existing server private key.
+		return logObserver.FilterMessage("Failed to load certificate").
+			FilterField(zap.String("certificate", certFile.Name())).Len() > 0
+	}, 100, time.Millisecond*200)
+
+	assert.True(t, logObserver.
+		FilterMessage("Failed to load certificate").
+		FilterField(zap.String("certificate", certFile.Name())).Len() > 0,
+		"Failed to find wanted logs. All logs: "+fmt.Sprint(logObserver.All()))
```

Comment:
One minor point - could you differentiate this log message from the one at the end of this function - e.g. "Unable to locate 'Failed to load certificate' error in log"?

Reply:
Yup, good pickup, @objectiser.